master/internal/api_notebook.go

```diff
@@ -27,3 +34,46 @@ func (a *apiServer) KillNotebook(
 	_ context.Context, req *apiv1.KillNotebookRequest) (resp *apiv1.KillNotebookResponse, err error) {
 	return resp, a.actorRequest(fmt.Sprintf("/notebooks/%s", req.NotebookId), req, &resp)
 }
+
+func (a *apiServer) NotebookLogs(
+	req *apiv1.NotebookLogsRequest, resp apiv1.Determined_NotebookLogsServer) error {
+	if err := grpc.ValidateRequest(
+		grpc.ValidateLimit(req.Limit),
+	); err != nil {
+		return err
+	}
+
+	cmdManagerAddr := actor.Addr("notebooks", req.NotebookId)
+	eventManager := a.m.system.Get(cmdManagerAddr.Child("events"))
+
+	logRequest := api.LogsRequest{
+		Offset: int(req.Offset),
+		Limit:  int(req.Limit),
+		Follow: req.Follow,
+	}
+
+	onLogEntry := func(log *logger.Entry) error {
+		return resp.Send(&apiv1.NotebookLogsResponse{LogEntry: api.LogEntryToProtoLogEntry(log)})
+	}
+
+	logStreamActorAddr := cmdManagerAddr.Child("logStream-" + string(rand.Int()))
+	logStreamActor, created := a.m.system.ActorOf(
+		logStreamActorAddr,
+		api.NewCommandLogStreamActor(
+			resp.Context(),
+			eventManager,
+			logRequest,
+			onLogEntry,
+		),
+	)
+
+	if !created {
+		// Either there is a collision in actor address or actor creation failed.
```

Comment:
hmm maybe a better place for it would be the ActorOf function itself. it isn't called out what that bool is supposed to represent (there is a named var assigned to it at on code branch but..).

Reply:
i removed and commented on the return value of ActorOf